packages/client/src/app/pages/forms/edit/builder/tabs/layout-editor/property-editor/form-controls/options-editor/custom-options.tsx

```diff
@@ -0,0 +1,292 @@
+
+    options.push({
+      label: '',
+      value: '',
+      default: false,
+      autofocus: true,
+    });
+
+    if (onChange) {
+      onChange({
+        ...value,
+        options,
+      });
+    }
+  };
+
+  /**
+   * Filters out the option based on its index
+   * @param optionIndex
+   */
+  const deleteOption = (optionIndex: number): void => {
+    let options = JSON.parse(JSON.stringify(value.options));
+
+    options = options.filter(
+      (option: OptionProp, index: number) => index !== optionIndex
+    );
+
+    if (onChange) {
+      onChange({
+        ...value,
+        options,
+      });
+    }
+  };
+
+  const dragAndDropOption = (optionIndex: number): void => {
+    // TODO: Implement DnD once we have added DnD functionality elsewhere
+  };
+
+  /**
+   * Find and update option property value
+   * @param payload
+   */
+  const updateOption = (payload: PayloadProp): void => {
+    const options = JSON.parse(JSON.stringify(value.options));
+
+    options.forEach((option: OptionProp, index: number) => {
+      if (index === payload.index) {
+        if (payload.key === 'label') {
+          option['label'] = String(payload.value);
+
+          if (!value.useCustomValues) {
+            option['value'] = String(payload.value);
+          }
+        } else if (payload.key === 'value') {
+          option['value'] = String(payload.value);
+        } else {
+          // Unsets for all other options
+          options.forEach(
+            (otherOption: OptionProp) => (otherOption.default = false)
+          );
+
+          // Sets the current option
+          option['default'] = Boolean(payload.value);
+        }
+      }
+    });
+
+    if (onChange) {
+      onChange({
+        ...value,
+        options,
+      });
+    }
+  };
+
+  // Add new option when the enter key is pressed
+  useEffect(() => {
+    if (enterKeyPress) {
+      addOption();
+    }
+  }, [enterKeyPress]);
+
+  return (
+    <Wrapper>
+      <Row>
+        <Column>
+          <H3>Options</H3>
+          <CheckboxWrapper>
+            <CheckboxInput
+              id="useCustomValues"
+              label="Use custom values"
+              checked={(value.useCustomValues as boolean) || false}
+              onChange={() =>
+                onChange &&
+                onChange({
+                  ...value,
+                  useCustomValues: !value.useCustomValues,
+                })
+              }
+            />
+          </CheckboxWrapper>
+        </Column>
+      </Row>
+      <Row>
+        <Column>
+          <TableOptions>
+            <thead>
+              <tr>
+                <td>Label</td>
+                {value.useCustomValues && <td>Value</td>}
+                <td></td>
+                <td></td>
+                <td></td>
+              </tr>
+            </thead>
+            <tbody>
+              {value.options &&
+                value.options.map((option: OptionProp, index: number) => (
```

Comment:
I think you should be using `options` instead, which already has the correct type and a default value set up.

Also you can type it better and not have to type the `.map()` arguments.
```suggestion
              {options.map((option, index) => (
```